rustls/src/vecbuf.rs

```diff
@@ -128,14 +145,19 @@ impl ChunkVecBuffer {
         Ok(offs)
     }
 
-    fn consume(&mut self, mut used: usize) {
-        while let Some(mut buf) = self.chunks.pop_front() {
-            if used < buf.len() {
-                buf.drain(..used);
+    fn consume(&mut self, used: usize) {
+        // first, mark the rightmost extent of the used buffer
+        self.prefix_used += used;
+
+        // then reduce `prefix_used` by discarding wholly-covered
+        // buffers
+
+        while let Some(buf) = self.chunks.pop_front() {
```

Comment:
We currently do:

- `pop_front()`, check, then `push_front()` if necessary

Do you think it might be faster to do instead inspect front in place and only pop it if necessary?